src/lib/syncPagePath.ts

```diff
@@ -121,6 +133,17 @@ async function pushPage(
       }
     }
 
+    if (this.route === 'custom_pages') {
+      const customPagePayload = { ...payload } as CustomPagesRequestRepresentation;
```

Comment:
this feels a tad safer given how nested this object may be:

```suggestion
      const customPagePayload = structuredClone(payload) as CustomPagesRequestRepresentation;
```